src/css/components/icon.scss

```diff
@@ -24,16 +24,27 @@
 .icon {
   display: inline-block;
   fill: transparent;
-  height: 1em;
-  max-height: 2rem;
-  max-width: 2rem;
-  min-height: 20px;
-  min-width: 20px;
-  stroke: currentColor;
   position: relative;
+  stroke: currentColor;
   top: -0.0625rem;
   vertical-align: middle;
   width: 1em;
+
+  &-small {
+    @extend .icon;
+    height: $grid-2;
+    width: $grid-2;
+  }
+  &-medium {
+    @extend .icon;
+    height: $grid-3;
+    width: $grid-3;
+  }
+  &-large {
+    @extend .icon;
```

Comment:
This isn't in the styleguide, because when I wrote it, it wasn't a commonly used thing in scss, but I also don't think I want to use the `&-name` functionality in scss because it makes things hard to search for. I want to be able to search for `icon-small` in the whole codebase and find it.

In terms of the `@extend`, I would expect you'd have to use both `icon` and `icons-{size}` classes to get an icon to render based on size (which is how most of the exiting code works. So I'd prefer not extending and requiring devs to use both classes.

I also had separate comments on having icons sizes, I think we resolved that here: https://github.com/18F/cg-dashboard/pull/853.